devolving_music/lib/song_scores.py

```diff
@@ -29,18 +31,35 @@ def get_scores(self) -> "dict[int, ScoreSuite]":
 
         return self.song_score_dict
 
+    def get_scores_list(self) -> "list[ScoreSuite]":
+        # calculates scores for all submissions using current comparisons
+        # returns list of all score suite objects
+        return list(self.get_scores().values())
+
     def get_compare_submission_random(self, submission_id) -> SongSubmission:
         key_list = list(self.song_score_dict.keys())
         key_list.remove(submission_id)
-        # once you have a critical number of comparisons then pull from quality
-        # list
-        return self.song_score_dict.get(random.choice(key_list)).song_submission
+        return self.song_score_dict.get(
+            random.choice(key_list)).song_submission
+
+    def get_compare_submission_closest(self, score_suite_obj: "ScoreSuite", information_threshold : int, update_scores = False):
+        # information threshold is the threshhold at which we start to consider score_suites as being able to be close to score_suite_obj
+        # score_suites below this information threshhold are considered to be infinitely far away from score_suite_obj
+        if(update_scores):
+            self.get_scores()
+        score_dict = copy.deepcopy(self.song_score_dict)
+        target_id = score_suite_obj.song_submission.id
+        score_dict.pop(target_id)
+        score_list=list(score_dict.values())
```

Comment:
```suggestion
        [
          score_suite
          for id, score_suite in self.song_score_dict.items()
          if id != score_suite_obj.id
        ]
```

A comprehension is much more readable and straightforward.

Reply:
I disagree about this I normally don't find comprehensions more readable. It's very clear what each line does step by step in the comprehension I have to think more about what is going on.  I'd agree if I had a for loop but pop is far more readable than a list comprehension.